src/Symfony/Component/Intl/Transliterator/EmojiTransliterator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Intl\Transliterator;
+
+final class EmojiTransliterator
+{
+    private static array $transliteratorByLocale = [];
+
+    public static function getInstance(string $locale): \Transliterator
+    {
+        return self::$transliteratorByLocale[$locale] ??= self::createTransliterator($locale);
+    }
+
+    private static function createTransliterator(string $locale): \Transliterator
+    {
+        $rulesFilename = __DIR__."/../Resources/data/transliterator/emoji/$locale.txt";
+        if (!file_exists($rulesFilename)) {
+            throw new \RuntimeException(sprintf('The transliteror rules source does not exist for locale "%s".', $locale));
```

Comment:
```suggestion
            throw new \RuntimeException(sprintf('The transliterator rules source does not exist for locale "%s".', $locale));
```
transliteror -> transliterator